clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -86,6 +85,25 @@ public AbortedTransaction(long pid, long firstOffset) {
         }
 
         @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+
+            AbortedTransaction that = (AbortedTransaction) o;
+
+            return pid == that.pid && firstOffset == that.firstOffset;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = (int) (pid ^ (pid >>> 32));
```

Comment:
This is the conventional style of `equals` we use everywhere else, so it seems fine to follow it here. I doubt anyone stresses much over its "complexity" since it is basically generated code.